cargo-pgrx/src/command/upgrade.rs

```diff
@@ -0,0 +1,216 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+use cargo_edit::{registry_url, Dependency};
+use eyre::eyre;
+use std::path::{Path, PathBuf};
+use toml_edit::KeyMut;
+use tracing::{debug, error, info, warn};
+
+use crate::CommandExecute;
+
+/// Upgrade pgrx crate versions in `Cargo.toml`.
+/// Defaults to latest.
+#[derive(clap::Args, Debug)]
+#[clap(author)]
+pub(crate) struct Upgrade {
+    /// Specify a version to upgrade to, rather than defaulting to the latest
+    /// version.
+    #[clap(long, short)]
+    pub(crate) target_version: Option<String>, //TODO: typed version not string
```

Comment:
oh interesting - so here we don't have a direct analogue to `cargo-upgrade`'s flag, it uses `-p, --package <PKGID[@<VERSION>]>`, but we already know we're upgrading all the pgrx crates at once, so -p doesn't make sense for us.

i wonder if it makes sense to just have an optional positional argument here instead? so instead of `cargo pgrx upgrade -t 0.12` it would be `pgrx upgrade 0.12`. just a little quality of life thing.

Reply:
After some discussion with Jubilee, we went with `--to`